File: 1637. Widest Vertical Area Between Two Points Containing No Points.py
class Solution:
    # O(n log n) time and O(1 space) complexity
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        points = sorted(points, key=lambda x: x[0])

        ret = 0
        for i in range(1, len(points)):
            diff = points[i][0] - points[i - 1][0]
            ret = max(ret, diff)

        return ret

File: 1637. Widest Vertical Area Between Two Points Containing No Points.cpp
class Solution {
public:
    // O(n log n) time and O(1) space complexity
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        sort(points.begin(), points.end());

        int ret = 0;
        for (int i = 1; i < points.size(); i++) {
            int diff = points[i][0] - points[i - 1][0];
            ret = max(ret, diff);
        };

        return ret;
    };
};
